src/main/java/school/hei/haapi/repository/dao/UserManagerDao.java

```diff
@@ -50,11 +52,58 @@ public List<User> findByCriteria(User.Role role, String ref, String firstName, S
         .where(builder.and(hasUserRole, hasUserRef, hasUserFirstName, hasUserLastName))
         .orderBy(QueryUtils.toOrders(pageable.getSort(), root, builder));
 
+
     return entityManager.createQuery(query)
         .setFirstResult((pageable.getPageNumber()) * pageable.getPageSize())
         .setMaxResults(pageable.getPageSize())
         .getResultList();
   }
+    public List<User> findByPromotionName(User.Role role, Pageable pageable,
```

Comment:
For now, we can use it. 

But add a TODO where we handle filter promotions by date intervals.